test/distributed/fsdp/test_fsdp_tp_integration.py

```diff
@@ -0,0 +1,285 @@
+# Owner(s): ["oncall: distributed"]
+
+import sys
+
+import torch
+from torch import distributed as dist
+from torch.distributed._shard import shard_module
+from torch.distributed._shard.sharded_optim import (
+    named_params_with_sharded_tensor,
+    ShardedOptimizer,
+)
+from torch.distributed._shard.sharding_plan import ShardingPlan
+from torch.distributed._shard.sharding_spec import ChunkShardingSpec
+from torch.distributed.fsdp.fully_sharded_data_parallel import (
+    CPUOffload,
+    FullyShardedDataParallel as FSDP,
+)
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+    TEST_WITH_DEV_DBG_ASAN,
+)
+
+
+if not dist.is_available():
+    print("Distributed not available, skipping tests", file=sys.stderr)
+    sys.exit(0)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+LR = 3e-5
+
+
+OPS_NOT_SHARD = [
+    "net3.weight",
+    "net3.bias",
+]
+
+SHARD_PARAMS = [
+    "net1.weight",
+    "net2.weight",
+]
```

Comment:
Curious: does it matter that not all parameters in `SimpleModel` appear in either of these two lists?

Reply:
It should be fine. This is just for grad sync. For real use case, we don't have something like this.